oxidation/crates/parsec_client_types/schema/local_workspace_manifest.json

```diff
@@ -19,19 +19,40 @@
             "type": "Map<EntryName, EntryID>"
         },
         {
+            // Added in Parsec v1.15
+            // Confined entries are entries that are meant to stay locally and not be added
+            // to the uploaded remote manifest when synchronizing. The criteria for being
+            // confined is to have a filename that matched the "prevent sync" pattern at the time of
+            // the last change (or when a new filter was successfully applied)
             "name": "local_confinement_points",
             "type": "Set<EntryID>",
             "introduced_in_revision": 15
         },
         {
+            // Added in Parsec v1.15
+            // Filtered entries are entries present in the base manifest that are not exposed
+            // locally. We keep track of them to remember that those entries have not been
+            // deleted locally and hence should be restored when crafting the remote manifest
+            // to upload.
             "name": "remote_confinement_points",
             "type": "Set<EntryID>",
             "introduced_in_revision": 15
         },
         {
+            // Added in Parsec v1.15
+            // Speculative placeholders are created when we want to access a workspace
+            // but didn't retrieve manifest data from backend yet. This implies:
+            // - non-placeholders cannot be speculative
+            // - the only non-speculative placeholder is the placeholder initialized
+            //   during the initial workspace creation
+            // This speculative information is useful during merge to understand if
+            // a data is not present in the placeholder compared with a remote because:
+            // a) the data is not locally known (speculative is True)
+            // b) the data is known, but has been locally removed (speculative is False)
+            // Prevented to be `required=True` by backward compatibility
             "name": "speculative",
             "type": "Boolean",
-            "introduced_in_revision": 6
+            "introduced_in_revision": 15
```

Comment:
Watch out `revision 15` != `Parsec v1.15` !!!

There is no revision logic on the data, only on the protocol stuff (the rational is, unlike in protocol where client and server can evolve together out of an old format)

However we cannot just remove the `introduced_in_revision` field here (given it is used to mark that the field is maybe present), so you should just set `introduced_in_revision: 115` and add a comment `// hack to indicate this field was added in Parsec v1.15, given revision is meaningless for data`

Reply:
and can you check for other place in the data where `introduced_in_revision` should be corrected

Reply:
and don't worry for the hack thing, I'm currently writing some documentation on api versioning strategy, so we will fix this when this is ready ;-)

Reply:
Done